src/fsharp/TypeChecker.fs

```diff
@@ -8714,7 +8714,7 @@ and TcMethodApplication
     let denv = env.DisplayEnv
 
     let isSimpleFormalArg (isParamArrayArg, isOutArg, optArgInfo: OptionalArgInfo, _callerInfoInfo: CallerInfoInfo, _reflArgInfo: ReflectedArgInfo) = 
-        not isParamArrayArg && not isOutArg && not optArgInfo.IsOptional 
+        not isParamArrayArg && not isOutArg && not optArgInfo.IsOptional && _callerInfoInfo = NoCallerInfo
```

Comment:
@AviAvni please remove the `_` from the variable name, since it is now being used. `_` is added only to suppress the "unused variable" warning.